### PR TITLE
♿ Add tabindex to lovelace elements

### LINKED ISSUE #4159: Add tabindex to Lovelace elements
- [ ] thermostat-card: slider (not sure if interaction with it is possible yet, but should be able to use arrow keys)
- [x] thermostat-card: mode icons
- [ ] light-card: slider
- [x] light-card: icon
- [x] glance-card: state-badges
- [x] entity-button: button
- [ ] media-player: (want to do a rewrite before we tackle this)
- [x] picture-elements: all the elements
- [x] picture-glance: image
- [x] picture-glance: icons
- [x] picture-entity: image
- [x] plant-status: haven't verified as it is not on the demo, but assume no values are tabbable
- [x] shopping-list-card: clear items icon
- [x] sensor-card: card itself
- [x] weather-forecast: card itself
- [x] state-label-badge

### PR DESCRIPTION
**Pull request by iantrich**
partially fixes https://github.com/home-assistant/home-assistant-polymer/issues/4159

blocked by https://github.com/home-assistant/home-assistant-polymer/pull/4115

~~need to look at sliders still~~ @thomasloven has added this in 0.3.2 of roundslider

### CODE REVIEW

**Comment by iantrich** [comment]
Last commit added this to picture-glance icons
image
Versus this
image

I think it looks a lot better. Color should probably be part of the theme vars. Trying to determine where this should all apply; mostly inner element, I think. e.g. doing this to the entity-button doesn't make sense

**Comment by dshokouhi** [comment]
@iantrich focus indicator needs to be anything actionable, so if entity button is a button it definitely needs it: https://www.w3.org/TR/UNDERSTANDING-WCAG20/navigation-mechanisms-focus-visible.html

The color needs to have proper contrast as well, so we should make sure the default theme aligns properly.

**Comment by iantrich** [comment]
I'm was only referring to the circular border radius

File: src/panels/lovelace/cards/hui-picture-glance-card.ts [hunk 1]
```diff
@@ -318,6 +318,11 @@ class HuiPictureGlanceCard extends LitElement implements LovelaceCard {
         padding-bottom: 4px;
         padding-top: 4px;
       }
+      ha-icon:focus {
+        outline: none;
+        box-shadow: 0 0 2px 2px #51a7e8;
+        border-radius: 100px;
```
Comment (bramkragten) [inline src/panels/lovelace/cards/hui-picture-glance-card.ts]: We should stay closer to material, so I suggest something like this: (not sure if it is the correct color and if we have a good variable for it)
```css
        outline: none;
        background: var(--divider-color);
```

Comment (iantrich) [reply src/panels/lovelace/cards/hui-picture-glance-card.ts]: image
yeah that looks close to what you get on mwc-button

Comment (iantrich) [reply src/panels/lovelace/cards/hui-picture-glance-card.ts]: haven't come up with anything good looking for cards or state-label-badge (found in badges, glance and picture-elements) yet

**Comment by iantrich** [comment]
laptop is about to die, but have state-label-badge and images left

**Comment by iantrich** [comment]
I'm not going to address badges and images in this PR

**Review by bramkragten — APPROVED** [review]
Looks good, just the badges might need some styling love:
image

I suggest:
image

```css
      ha-state-label-badge:focus {
        outline: none;
        background: var(--divider-color);
        border-radius: 4px;
      }
      ha-state-label-badge {
        display: inline-block;
        padding: 4px;
        margin: -4px 0 -4px 0;
      }
```

**Comment by bramkragten** [comment]
And we should add a new variable instead of (ab)using `--divider-color`

File: src/panels/lovelace/cards/hui-glance-card.ts [hunk 2]
```diff
@@ -167,6 +167,11 @@ export class HuiGlanceCard extends LitElement implements LovelaceCard {
         margin-bottom: 12px;
         width: var(--glance-column-width, 20%);
       }
+      .entity:focus {
+        outline: none;
+        background: var(--divider-color);
+        border-radius: 15%;
```
Comment (bramkragten) [inline src/panels/lovelace/cards/hui-glance-card.ts]: I think a 4px radius works better here.

Comment (bramkragten) [reply src/panels/lovelace/cards/hui-glance-card.ts]: And you might want to add a little padding:
```css
        padding: 4px;
        margin: -4px;
```

**Comment by iantrich** [comment]
> Looks good, just the badges might need some styling love:
> image
> 
> I suggest:
> image
> 
> ```css
>       ha-state-label-badge:focus {
>         outline: none;
>         background: var(--divider-color);
>         border-radius: 4px;
>       }
>       ha-state-label-badge {
>         display: inline-block;
>         padding: 4px;
>         margin: -4px 0 -4px 0;
>       }
> ```

I said I wasn't going to do badges in this PR 😛 but I'll take a look, I guess...

**Comment by bramkragten** [comment]
Then don't give it a tab-index 😝

**Review by bramkragten — APPROVED** [review]

**Pull request merged**